src/basics.md

```diff
@@ -1770,6 +1771,48 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-read-environment-variable]: #ex-read-environment-variable
+<a name="ex-read-environment-variable"></a>
+## Read environment variable
+[![std-badge]][std] [![cat-filesystem-badge]][cat-filesystem]
+
+Set an environment variable and read it back. If the variable is not found, then fallback
+to a default value.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+use std::env;
+use std::path::PathBuf;
+
+# error_chain! {
+#     foreign_links {
+#         Utf8(std::str::Utf8Error);
+#         AddrParse(std::net::AddrParseError);
+#     }
+# }
+
+fn set_env_variable() {
+  env::set_var("CONFIG_FILE", "config.toml");
+}
+
+fn run() -> Result<()> {
+    set_env_variable();
```

Comment:
setting and reading the same env variable does not make a control flow that is close to being realistic. How about we set one variable and read another?